cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/NotificationSubscriberService.java

```diff
@@ -87,24 +82,22 @@
   private final DatasetFramework datasetFramework;
   private final MultiThreadDatasetCache multiThreadDatasetCache;
   private final CConfiguration cConf;
-  private final Store store;
   private ListeningExecutorService taskExecutorService;
   private volatile boolean stopping = false;
 
 
   @Inject
   NotificationSubscriberService(MessagingService messagingService,
-                                CConfiguration cConf, Store store,
+                                CConfiguration cConf,
                                 DatasetFramework datasetFramework,
                                 TransactionSystemClient txClient) {
     this.cConf = cConf;
-    this.store = store;
     this.messagingContext = new MultiThreadMessagingContext(messagingService);
     this.multiThreadDatasetCache = new MultiThreadDatasetCache(
       new SystemDatasetInstantiator(datasetFramework), txClient,
       NamespaceId.SYSTEM, ImmutableMap.<String, String>of(), null, null, messagingContext);
     this.transactional = Transactions.createTransactionalWithRetry(
-      Transactions.createTransactional(multiThreadDatasetCache),
+      Transactions.createTransactional(multiThreadDatasetCache, Schedulers.SUBSCRIBER_TX_TIMEOUT_SECONDS),
```

Comment:
Why not continue to have subscriber service use the system-wide transaction timeout?